cdap-ui/app/cdap/components/PreviewData/Table.tsx

```diff
@@ -97,6 +97,13 @@ const DataTableView: React.FC<IDataTableProps> = ({
     return msg;
   };
 
+  const format = (field: any) => {
+    if (typeof field === 'object') {
```

Comment:
might have to test this with undefined and null values to make sure it renders ok

Reply:
I think either should be ok as react can render both null and undefined. Although in this case it will render empty value and "null" as string

Reply:
Both null and undefined get rendered as empty values because the backend returns undefined for both (even if the value was null in the original dataset).

Reply:
More specifically: For fields that have values of undefined or null, the backend doesn't return anything in the record for that field, so when the UI checks for the value of the field in that record, it returns "undefined."